echo-extended/server/echo-extended.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 Cartesi Pte. Ltd.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy of the
+# License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+
+import json
+import logging
+from os import environ
+
+import requests
+from flask import Flask, request
+
+app = Flask(__name__)
+app.logger.setLevel(logging.INFO)
+
+dispatcher_url = environ["HTTP_DISPATCHER_URL"]
+app.logger.info(f"HTTP dispatcher url is {dispatcher_url}")
+
+
+def unserialize(data):
+    raw_data = bytes.fromhex(data).decode("utf-8")
+    data = json.loads(raw_data)
+
+    return data
+
+
+def serialize(data):
+    return "0x" + data.encode("utf-8").hex()
+
+
+@app.route("/advance", methods=["POST"])
+def advance():
+    transformations = {
+        "upper": str.upper,
+        "lower": str.lower,
+    }
+
+    body = request.get_json()
+
+    payload = unserialize(body["payload"][2:])
+    app.logger.info("[APP] Unserialized payload: " + payload)
+
+    transform = payload["transform"]
+    message = payload["message"]
+
+    # Error path
+    if transform not in transformations:
+        error_message = "Cannot do the '" + transform + "' transformation"
+        error_serialize = serialize(error_message)
+        requests.post(dispatcher_url + "/notice", json={"payload": error_message})
+        requests.post(dispatcher_url + "/finish", json={"status": "accept"})
+
+        app.logger.info("[ERROR PATH] Error message: " + error_message)
+        app.logger.info("[APP] Error message serialized: " + error_serialize)
+
+        return "", 202
```

Comment:
We haven't touched error handling yet in our documentation or examples, so this is a very good opportunity to start! We could extend the `extended-echo` DApp to also become the first error handling example.

In that regard, as this is a simple recoverable error (wrong input), the way to go would be to issue a [report](https://github.com/cartesi/rollups/blob/main/openapi/dapp.yaml#L157) and call `/finish` with `reject` instead. Reports are preferred in error scenarios because they persist even if the input is rejected, which is not the case with notices and vouchers, who get discarded.